src/NexusMods.App.UI/Pages/LibraryPage/NexusModsModPageLibraryItemModel.cs

```diff
@@ -32,10 +36,29 @@ public NexusModsModPageLibraryItemModel(IObservable<IChangeSet<NexusModsLibraryI
 
         // ReSharper disable once NotDisposedResource
         var datesDisposable = ILibraryItemWithDates.SetupDates(this);
+        var modPageThumbnailPipeline = ImagePipelines.GetModPageThumbnailPipeline(serviceProvider);
 
         // NOTE(erri120): This subscription needs to be set up in the constructor and kept alive
         // until the entire model gets disposed. Without this, selection would break for off-screen items.
-        var libraryItemsDisposable =  libraryItemsObservable.OnUI().SubscribeWithErrorLogging(changeSet => LibraryItems.ApplyChanges(changeSet));
+        // ReSharper disable once AsyncVoidLambda
+        var libraryItemsDisposable =  libraryItemsObservable.OnUI().SubscribeWithErrorLogging(async changeSet =>
+        {
+            LibraryItems.ApplyChanges(changeSet);
+            
+            // Note(sewer): Update the thumbnail of the header to be the thumbnail of the first child item.
+            // By definition, all the sub items belong to this page, so the thumbnail of the child item
+            // is the thumbnail of this page.
+            //
+            // SAFETY: This can't have race condition, code is executed on UI, so can't be executed in parallel.
+            var first = LibraryItems.FirstOrDefault();
+            if (!first.IsValid()) return;
+            try
+            {
+                var thumbNail = await modPageThumbnailPipeline.LoadResourceAsync(first.ModPageMetadataId, CancellationToken.None);
+                Thumbnail.Value = thumbNail.Data.Value;
+            }
+            catch (Exception) { /* suppress unhandled error in case lookup fails silently */ }
+        });
```

Comment:
I hate this, we're doing async/await inside a void lambda. You can do something like this instead:

```csharp
OnUI().Select(changeSet => {
  LibraryItems.ApplyChanges(changeSet);
  return LibraryItems.FirstOrDefault();
}).Where(item => item.IsValid())
.SelectMany(async item => /* load */).SubscribeWithErrorLogging();
```